libs/wingc/src/lib.rs

```diff
@@ -116,13 +120,18 @@ pub fn parse(source_file: &str) -> (Scope, Diagnostics) {
 	let tree = match parser.parse(&source[..], None) {
 		Some(tree) => tree,
 		None => {
-			panic!("Failed parsing source file: {}", source_file);
+			panic!("Failed parsing source file: {}", source_path.display());
 		}
 	};
 
 	let wing_parser = Parser {
 		source: &source[..],
-		source_name: source_file.to_string(),
+		source_name: source_path
```

Comment:
Updated

Reply:
Actually I've decided to leave for the future (the parser is only using the immediate file name, not the full path, so it would just add a lot of unnecessary clones at the moment). I think this should be done in a separate refactoring which adds a `Parser::new` method etc.